action-server/actions/question_answering_form.py

```diff
@@ -0,0 +1,87 @@
+import os
+from typing import Any, Dict, List, Text, Union
+
+from rasa_sdk import Tracker
+from rasa_sdk.events import SlotSet
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.forms import FormAction
+
+from actions.answers import (
+    QuestionAnsweringProtocol,
+    QuestionAnsweringResponse,
+    QuestionAnsweringStatus,
+)
+
+FAQ_URL_ENV_KEY = "COVID_FAQ_SERVICE_URL"
+DEFAULT_FAQ_URL = "https://covidfaq.dialoguecorp.com"
+
+
+class QuestionAnsweringForm(FormAction):
+    def name(self) -> Text:
+
+        return "question_answering_form"
+
+    @staticmethod
+    def required_slots(tracker: Tracker) -> List[Text]:
+        status = tracker.get_slot("question_answering_status")
+        if status == QuestionAnsweringStatus.SUCCESS:
+            return ["feedback"]
+
+        return ["active_question"]
+
+    def slot_mappings(self) -> Dict[Text, Union[Dict, List[Dict]]]:
+        return {
+            "active_question": self.from_text(),
+            "feedback": [
+                self.from_intent(intent="affirm", value=True),
+                self.from_intent(intent="deny", value=False),
+            ],
+        }
+
+    def validate_active_question(
+        self,
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Dict[Text, Any]:
+        protocol = QuestionAnsweringProtocol(
+            os.environ.get(FAQ_URL_ENV_KEY, DEFAULT_FAQ_URL)
+        )
+        result: QuestionAnsweringResponse = protocol.get_response(value)
+
+        if result.status == QuestionAnsweringStatus.SUCCESS and result.answers:
+            dispatcher.utter_message(result.answers[0])
+
+        return {"question_answering_status": result.status, "answers": result.answers}
```

Comment:
I think you should use constants for slots since here it might not be clear that these are slot names. The constant names are reused at different places.